### ISSUE TITLE
energy_new in CAM

### ISSUE THREAD

**Issue by eugenelawrence** [junfu1115/DANet#9]
Hi @junfu1115 ,
why do you use the energy_new in [attention.py](https://github.com/junfu1115/DANet/blob/master/encoding/nn/attention.py#L75)?
Correct me if I miss something.

**Comment by weix-liu** [comment]
To prevent overflow or underflow, it should be x - max(x), rather than max(x) - x

**Comment by HanzhiC** [comment]
Yes. I am also wondering why using max(x) - x, seems that your correlation will favor the most different channel information instead the most similar one

**Comment by kamiLight** [comment]
> Yes. I am also wondering why using max(x) - x, seems that your correlation will favor the most different channel information instead the most similar one

i have the same opinion with you, using max(x) - x will give bigger weights to the feature with less similarity after performing softmax

**Comment by carolo44** [comment]
> Yes. I am also wondering why using max(x) - x, seems that your correlation will favor the most different channel information instead the most similar one

I think it should be x - max(x) and actually this operation is also redundant,  pytorch itself has done it already.
